mediators/datamapper/components/org.wso2.carbon.gateway.mediators.datamapper.engine/src/main/java/org/wso2/carbon/gateway/mediators/datamapper/engine/core/models/MapModel.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.gateway.mediators.datamapper.engine.core.models;
+
+import java.util.Map;
+
+/**
+ * This class implements {@link Model} interface with a {@link Map} to hold data.
+ */
+public class MapModel implements Model<Map<String, Object>> {
```

Comment:
Any reason why Map<String,Object> ? (type of values in map is totally unknown for this part of the code?)?